pkg/scheduler/generic_scheduler.go

```diff
@@ -461,29 +459,41 @@ func prioritizeNodes(
 					metrics.SchedulerGoroutines.WithLabelValues(metrics.PrioritizingExtender).Dec()
 					wg.Done()
 				}()
+				nodeScoreList := scoresMap[extenderPrefix+extenders[extIndex].Name()]
 				prioritizedList, weight, err := extenders[extIndex].Prioritize(pod, nodes)
 				if err != nil {
 					// Prioritization errors from extender can be ignored, let k8s/other extenders determine the priorities
-					klog.V(5).InfoS("Failed to run extender's priority function. No score given by this extender.", "error", err, "pod", klog.KObj(pod), "extender", extenders[extIndex].Name())
 					return
 				}
-				mu.Lock()
 				for i := range *prioritizedList {
-					host, score := (*prioritizedList)[i].Host, (*prioritizedList)[i].Score
+					score := (*prioritizedList)[i].Score
 					if klogV.Enabled() {
-						klogV.InfoS("Extender scored node for pod", "pod", klog.KObj(pod), "extender", extenders[extIndex].Name(), "node", host, "score", score)
+						klogV.InfoS("Extender scored node for pod", "pod", klog.KObj(pod), "extender", extenders[extIndex].Name(), "node", nodeScoreList[i].Name, "score", nodeScoreList[i].Score)
 					}
-					combinedScores[host] += score * weight
+					mu.Lock()
```

Comment:
Each routine is accessing a different slice, so there is no need for a mutex anymore.